homeassistant/components/proximity/__init__.py

```diff
@@ -47,12 +48,20 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     hass.data.setdefault(DOMAIN, {})
     for zone, proximity_config in config[DOMAIN].items():
         _LOGGER.debug("setup %s with config:%s", zone, proximity_config)
+
         coordinator = ProximityDataUpdateCoordinator(hass, zone, proximity_config)
-        await coordinator.async_config_entry_first_refresh()
+        await coordinator.async_refresh()
         hass.data[DOMAIN][zone] = coordinator
+
         proximity = Proximity(hass, zone, coordinator)
-        proximity.async_write_ha_state()
         await proximity.async_added_to_hass()
+        proximity.async_write_ha_state()
+
+        async_track_state_change(
```

Comment:
Just noting that we've now moved the start of the tracking after the first coordinator refresh. Previously it was done before the first refresh.

Reply:
just to keep the logic as is, i'll move it before the first coordinator fetch

Reply:
Regardless of this, we still need some logic for inital data fetch (_was also not present in the old code_) in case the proximity integration is loaded after an integration, which provides the to be tracked devices, those no state change will be there, unless the tracked devices change their state/location